cmake/developer_package/compile_flags/functions.cmake

```diff
@@ -479,3 +479,38 @@ function(ov_target_link_libraries_as_system TARGET_NAME LINK_TYPE)
         set_target_properties(${TARGET_NAME} PROPERTIES NO_SYSTEM_FROM_IMPORTED ON)
     endif()
 endfunction()
+
+#
+# Detects the standard filesystem library and sets the OPENVINO_STD_FS_LIB variable with library name if required.
+#
+function(ov_detect_std_filesystem_lib)
+    if(NOT DEFINED OPENVINO_STD_FS_LIB)
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+            "#include <experimental/filesystem>\nint main(int argc, char ** argv) {\n  std::experimental::filesystem::path p(argv[0]);\n  return p.string().length();\n}")
```

Comment:
do we really need to detect `experimental/filesystem` ? we currently require C++17 which enables filesystem support w/o experimental

Reply:
This part is removed.